nexus/reconfigurator/planning/src/reports.rs

```diff
@@ -0,0 +1,60 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Utilities for structured reports on planning, i.e., Blueprint generation.
+//!
+//! Most of the important structures (e.g., `PlanningReport`, the step reports)
+//! are defined in [`nexus_types::deployment`] so that they may be shared with
+//! the `blueprint_planner` background task and `omdb`.
+
+use nexus_types::deployment::{
+    PlanningAddStepReport, PlanningCockroachdbSettingsStepReport,
+    PlanningDecommissionStepReport, PlanningExpungeStepReport,
+    PlanningMgsUpdatesStepReport, PlanningNoopImageSourceStepReport,
+    PlanningReport, PlanningZoneUpdatesStepReport,
+};
+use omicron_uuid_kinds::BlueprintUuid;
+
+/// A blueprint planning report minus the blueprint ID that the
+/// report is for. Returned by [`crate::planner::Planner::do_plan`]
+/// when all planning steps are complete, but before the blueprint
+/// has been built (and so we don't yet know its ID).
+#[derive(Debug)]
+pub(crate) struct InterimPlanningReport {
```

Comment:
It looks like the blueprint_id is assigned when we call `BlueprintBuilder::build()`.  That can only be called once.  I wonder if we could remove this whole type (and this file) by generating the blueprint id when we call `BlueprintBuilder::new_based_on()` instead and making it available to the caller via a `new_blueprint_id()` method.  Not a big deal either way.